packages/core/nano/src/types.ts

```diff
@@ -281,6 +291,10 @@ export interface IMatrixReader<T> {
     readonly matrixProducer: IMatrixProducer<T>;
 }
 
+export interface IMatrixWriter<T> {
+    setCell(row: number, col: number, value: T): void;
+}
```

Comment:
Note that an IMatrixWriter cannot insert/remove rows/cols.  I'm currently using IVectorWriters to change the matrix shape.

Reply:
Good idea.  :-)